test/functional/mempool_accept.py

```diff
@@ -72,85 +69,71 @@ def run_test(self):
 
         self.log.info('A transaction already in the blockchain')
-        coin = coins.pop()  # Pick a random coin(base) to spend
-        raw_tx_in_block = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[{'txid': coin['txid'], 'vout': coin['vout']}],
-            outputs=[{node.getnewaddress(): 0.3}, {node.getnewaddress(): 49}],
-        ))['hex']
-        txid_in_block = node.sendrawtransaction(hexstring=raw_tx_in_block, maxfeerate=0)
+        raw_tx_in_block = miniwallet.send_self_transfer(from_node=node)
         node.generate(1)
         self.mempool_size = 0
         self.check_mempool_result(
-            result_expected=[{'txid': txid_in_block, 'allowed': False, 'reject-reason': 'txn-already-known'}],
-            rawtxs=[raw_tx_in_block],
+            result_expected=[{'txid': raw_tx_in_block['txid'], 'allowed': False, 'reject-reason': 'txn-already-known'}],
+            rawtxs=[raw_tx_in_block['hex']],
         )
 
         self.log.info('A transaction not in the mempool')
-        fee = Decimal('0.000007')
-        raw_tx_0 = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[{"txid": txid_in_block, "vout": 0, "sequence": BIP125_SEQUENCE_NUMBER}],  # RBF is used later
-            outputs=[{node.getnewaddress(): Decimal('0.3') - fee}],
-        ))['hex']
-        tx = tx_from_hex(raw_tx_0)
+        fee = Decimal("0.003")  # The default fee_rate in MiniWallet
+        coin = miniwallet.get_utxo()  # Get a utxo and store it for later
+        raw_tx_0 = miniwallet.create_self_transfer(fee_rate=fee, from_node=node, utxo_to_spend=coin, sequence=BIP125_SEQUENCE_NUMBER) # RBF is used later
+        tx = tx_from_hex(raw_tx_0['hex'])
         txid_0 = tx.rehash()
         self.check_mempool_result(
-            result_expected=[{'txid': txid_0, 'allowed': True, 'vsize': tx.get_vsize(), 'fees': {'base': fee}}],
-            rawtxs=[raw_tx_0],
+            result_expected=[{'txid': txid_0, 'allowed': True, 'vsize': tx.get_vsize(), 'fees': {'base': raw_tx_0['fees']}}],
+            rawtxs=[raw_tx_0['hex']],
         )
 
         self.log.info('A final transaction not in the mempool')
-        coin = coins.pop()  # Pick a random coin(base) to spend
-        output_amount = Decimal('0.025')
-        raw_tx_final = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[{'txid': coin['txid'], 'vout': coin['vout'], "sequence": 0xffffffff}],  # SEQUENCE_FINAL
-            outputs=[{node.getnewaddress(): output_amount}],
-            locktime=node.getblockcount() + 2000,  # Can be anything
-        ))['hex']
-        tx = tx_from_hex(raw_tx_final)
-        fee_expected = coin['amount'] - output_amount
+        raw_tx_final = miniwallet.create_self_transfer(from_node=node, sequence=0xffffffff, locktime=node.getblockcount()+2000) # SEQUENCE_FINAL, locktime can be anything
```

Comment:
stylistic nit, to better match comments to parameters:

Reply:
This was indeed how the first commit (aa068ad397cfe5766dccfd45c0f4ba846e7abd22) had it, but I think I missed it in later force pushes. Thank you!